### ISSUE TITLE
Fails to build on Python 3.10

### ISSUE THREAD

**Issue by keli5** [numpy/numpy#17044]
### Error message:
My terminal only goes so far back, so I can't get the full thing, but here's what I have: https://hastebin.com/bucomehava.coffeescript

### Numpy/Python version information:
Python: Python 3.10.0a0 (heads/master:39042e0, Aug 10 2020, 09:38:02)
I'm on Debian 10

**Comment by eric-wieser** [comment]
What version of numpy are you building?

**Comment by keli5** [comment]
I'm not exactly sure what version. I'm installing deps from a requirements.txt file that specifies `numpy ~= 1.18.4`

**Comment by keli5** [comment]
Oops - it appears this issue is with scipy. My bad, I didn't realize

**Comment by eric-wieser** [comment]
Well the release notes for the [latest 1.19.1 release](https://numpy.org/devdocs/release/1.19.1-notes.html) say:

> The Python versions supported for this release are 3.6-3.8

If you want to use newer versions of python, you'll need to install numpy from source.

**Comment by eric-wieser** [comment]
This isn't a scipy issue, the source files with the warnings are coming from numpy.

**Comment by keli5** [comment]
It's not? Numpy installed 1.19.1 fine with pip, but scipy throws an error and fails to install.

**Comment by eric-wieser** [comment]
Are you sure that the numpy 1.19 which installed just fine was being installed into a 3.10 python and not an older python? Can you show the output of `numpy.__version__` and `sys.version` in the same shell session?

**Comment by eric-wieser** [comment]
Well, maybe 1.19 does work on 3.10, but you're asking to install `1.18.4` which does not!

**Comment by keli5** [comment]
Here:
![image](https://user-images.githubusercontent.com/46039037/89800740-c9c18600-daf4-11ea-9fa0-b642dcf79c42.png)
```
Python 3.10.0a0 (heads/master:39042e0, Aug 10 2020, 09:38:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import sys
>>> numpy.__version__
'1.19.1'
>>> sys.version
'3.10.0a0 (heads/master:39042e0, Aug 10 2020, 09:38:02) \n[GCC 8.3.0]'
>>> 
```

**Comment by seberg** [comment]
Not sure that 1.19.1 should install. Its possible 1.19.1 works, we definitely will release a 1.19.x release which will support Python 3.9, but no guarantees for any 3.10 (especially its still in alpha, later releases will add incompatibilities probably).
The master branch may or may not be up to date, but PRs which fix it are very welcome there. Eventually, we will have to support 3.10 after all.

**Comment by keli5** [comment]
Since I'm using the same version of python for the installation of the requirements, it shouldn't try to install 1.18.4 as there's a working version of 1.19.1, yes?

**Comment by eric-wieser** [comment]
Your requirements file says `numpy ~= 1.18.4`, which means "I need 1.18.4, 1.18.5, ..., etc, but not 1.19.x". So I think pypi will try and downgrade numpy.

I suggest you update your requirements file to allow 1.19 too :)

**Comment by keli5** [comment]
Alright, though I still believe this is a scipy issue as it fails on scipy, I'll see if this changes anything

**Comment by keli5** [comment]
Installing gets this far:
![image](https://user-images.githubusercontent.com/46039037/89801950-69334880-daf6-11ea-93da-19d115438506.png)
and then errors afterwards, despite the updated requirements.txt including `numpy ~= 1.19.1.`

**Comment by eric-wieser** [comment]
Scipy is forcing an install of an old version of numpy here:

https://github.com/scipy/scipy/blob/master/pyproject.toml#L8
